### PR TITLE
Implement cache poisoning if error occurs during resolution stage

### PR DESCRIPTION
**Pull request by george-zubrienko**
## Scope

**Dependency updates:**

* Updated the Nexus service image in `docker-compose.yaml` from `1.1.5` to `1.1.7` and the Nexus Receiver image from `1.1.1` to `1.1.2` to ensure the latest features and bug fixes are used. [[1]](diffhunk://#diff-3fde9d1a396e140fefc7676e1bd237d67b6864552b6f45af1ebcc27bcd0bb6e9L42-R42) [[2]](diffhunk://#diff-3fde9d1a396e140fefc7676e1bd237d67b6864552b6f45af1ebcc27bcd0bb6e9L72-R72)

**Algorithm input validation and error handling:**

* Updated the main test suite to include a new test, `test_failing_reader`, which verifies that runs with negative `z` values are marked as failed and the error of type `NegativeZError` is recorded in the run metadata. [[1]](diffhunk://#diff-88d45d56ee2684ca0fc02f2a6c4f73b44ea97c8c5cf7b839fb286540b91340b3R45-R63) [[2]](diffhunk://#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R103-R112)

**SDK and typing improvements:**

* Fixed the decoding of the `lifecycle_stage` field in `scheduler.py` to ensure it is properly decoded from bytes if present.
* Implemented cache poisoning in case a run fails. [[1]](diffhunk://#diff-e016cbce5a01e9bccd02b7566c330990d3f1aac208bb31b77de95f35fe60dfcaL68-R68) [[2]](diffhunk://#diff-e016cbce5a01e9bccd02b7566c330990d3f1aac208bb31b77de95f35fe60dfcaR81-R85)

**Logging and telemetry:**

* Fixed incorrect logging of `else` branches for telemetry runs in `app_core.py`.

### CODE REVIEW

File: nexus_client_sdk/nexus/abstractions/algrorithm_cache.py [hunk 1]
```diff
@@ -78,10 +78,12 @@ def get_result(alias: str, completed_task: asyncio.Task) -> TResult:
...
         async def _execute(nexus_input: InputObject) -> TResult:
+            result: TResult | None = None
             async with nexus_input as instance:
-                result = await nexus_input.process(**kwargs)
-
-            self._cache[instance.cache_key()] = result
+                try:
+                    result = await nexus_input.process(**kwargs)
+                finally:
+                    self._cache[instance.cache_key()] = result
```
[the remark this answers is not in the document]
Comment (george-zubrienko) [reply nexus_client_sdk/nexus/abstractions/algrorithm_cache.py:86]: Invalid suggestion, this is working as intended

**Review by marcusforby — APPROVED** [review]
LGTM
